pkgs/tools/networking/openconnect/default.nix

```diff
@@ -9,15 +24,20 @@ let vpnc = fetchgit {
 };
 
 in stdenv.mkDerivation rec {
-  pname = "openconnect";
-  version = "8.10";
-
-  src = fetchurl {
-    urls = [
-      "ftp://ftp.infradead.org/pub/openconnect/${pname}-${version}.tar.gz"
-    ];
-    sha256 = "1cdsx4nsrwawbsisfkldfc9i4qn60g03vxb13nzppr2br9p4rrih";
-  };
+  pname = "openconnect${lib.optionalString head "-head"}";
+  version = if head then "2021-05-05" else "8.10";
+
+  src =
+    if head then fetchFromGitLab {
+      owner = "openconnect";
+      repo = "openconnect";
+      rev = "684f6db1aef78e61e01f511c728bf658c30b9114";
```

Comment:
That rev does not exist https://github.com/openconnect/openconnect/commit/684f6db1aef78e61e01f511c728bf658c30b9114

Reply:
openconnect is hosted on gitlab.com, not github.

https://gitlab.com/openconnect/openconnect/-/commit/684f6db1aef78e61e01f511c728bf658c30b9114